File: Python3/maximum_depth_of_N_ary_tree.py
"""
# Definition for a Node.
class Node:
    def __init__(self, val=None, children=None):
        self.val = val
        self.children = children
"""

class Solution:
    def maxDepth(self, root: 'Node') -> int:
        if not root: return 0
        
        n = len(root.children)
        result = 0
        
        for i in range(n):
            result = max(result, self.maxDepth(root.children[i]))
  
        return result + 1

File: C/maximum_depth_of_N_ary_tree.c
/**
 * Definition for a Node.
 * struct Node {
 *     int val;
 *     int numChildren;
 *     struct Node** children;
 * };
 */

int maxDepth(struct Node* root) {
    if(!root) return 0;

    int result = 0;
        
    for(int i = 0; i < root->numChildren; i++)
        result = fmax(result, maxDepth((root->children[i])));
    
    return result + 1;
}
